backend/routes/resetCourts.js

```diff
@@ -0,0 +1,97 @@
+const express = require("express");
+const router = express.Router();
+const admin = require("firebase-admin");
+const bcrypt = require("bcrypt");
+
+function bcryptCompare(plaintext, hashed) {
+  return new Promise(function (resolve, reject) {
+    bcrypt.compare(plaintext, hashed, function (err, res) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(res);
+      }
+    });
+  });
+}
+
+async function resetActivePlayers(location) {
+  const active_players = await location
+    .collection("activePlayers")
+    .listDocuments();
+
+  for (var j = 0; j < active_players.length; j++) {
+    const player_ref = await location
+      .collection("activePlayers")

```

Comment:
To optimize the number of reads and writes, can we consider the following:

Use a query to access the size of the active players - in order to find the number of courts at this location (since it can vary).

Then, for every element in the active queue, we call .set() to overwrite the data with our EmptyX player. This would avoid the individual reads and overwrite all players in a 1-n order:

const ref = db.collection("ActivePlayers").doc(player.id);
batch.set(ref, player);

Reply:
Hi Ron, thanks so much for this comment! When I originally tried using .set(), it did not change the ID of the document for each player (since the design doc specifies that the ID should be changed to Empty{x}); in my research, it seems that the only way to do this is to delete/rewrite the document with a new ID. Do you know of a different way that I can also change the document's ID? Thanks!

Reply:
Hey Ron, you brought up a great point. As such, I have realized that we can leave the ID of the document as fixed: "Court1", "Court2"... "Court{n}". This won't affect the functionality of other endpoints as we still store the firebaseUID as a property. Meaning, should we now be able to simply overwrite/set each document without changing its id?